src/protocol.h

```diff
@@ -390,35 +390,39 @@ class CAddress : public CService
     CAddress(CService ipIn, ServiceFlags nServicesIn) : CService{ipIn}, nServices{nServicesIn} {};
     CAddress(CService ipIn, ServiceFlags nServicesIn, NodeSeconds time) : CService{ipIn}, nTime{time}, nServices{nServicesIn} {};
 
-    SERIALIZE_METHODS(CAddress, obj)
+    enum class Format {
+        Disk,
+        Network,
+    };
+    struct SerParams : CNetAddr::SerParams {
+        const Format fmt;
+    };
```

Comment:
If you want to use designated initializers, a la `{.addrv2=false, .disk=false}`, then you can't use inheritance here, but could instead add an `operator CNetAddr::SerParams() const` function for pretty much the same result.

Reply:
Yeah, looks fine